staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -343,3 +345,31 @@ func addCapability(c *corev1.Container, capability corev1.Capability) {
 	}
 	c.SecurityContext.Capabilities.Add = append(c.SecurityContext.Capabilities.Add, capability)
 }
+
+// disallowPrivilegeEscalation configures the containers not allowed PrivilegeEscalation
+func disallowPrivilegeEscalation(p *corev1.Pod, containerName string) {
+	podutils.VisitContainers(&p.Spec, podutils.AllContainers, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		if c.SecurityContext == nil {
+			c.SecurityContext = &corev1.SecurityContext{}
+		}
+		c.SecurityContext.AllowPrivilegeEscalation = pointer.Bool(false)
+		return false
+	})
+}
+
+// setSeccompProfile apply SeccompProfile to the containers
+func setSeccompProfile(p *corev1.Pod, containerName string) {
+	podutils.VisitContainers(&p.Spec, podutils.AllContainers, func(c *corev1.Container, _ podutils.ContainerType) bool {
+		if c.Name != containerName {
+			return true
+		}
+		if c.SecurityContext == nil {
+			c.SecurityContext = &corev1.SecurityContext{}
+		}
+		c.SecurityContext.SeccompProfile = &corev1.SeccompProfile{Type: "RuntimeDefault"}
```

Comment:
This only works when the actual pod's `seccompProfile` is `RuntimeDefault`. When the actual pod's seccompProfile is `Localhost`, this won't work.

Reply:
@ardaguclu 
Restricted policy simply enforces the application of `seccompProfile`(`RuntimeDefault` or `Localhost`).
So even If  the actual pod's `seccompProfile` is `Localhost` and  EphemeralContainer is `RuntimeDefault`, it works fine complying with PSS(I have already checked this).
However, if there is any reason, I'll fix EphemeralContainer to use the same `seccompProfile`.

Reply:
Could you please add a test case in here https://github.com/kubernetes/kubernetes/blob/master/test/cmd/debug.sh for the cases where;
* restricted profile just works in not restricted namespace
* restricted profile works when pod's seccompProfile is `RuntimeDefault`
* restricted profile works when pod's seccompProfile is `Localhost`

Reply:
OK, I've added these test cases.

https://github.com/kubernetes/kubernetes/pull/117543/files#diff-61c3ff6b4ef690cba06269bfc4d3df8f92c0e991d6196db33843e16be45bf65f